### DISCUSSION TITLE
Devcontainer configuration for Codespaces

### DISCUSSION THREAD

**Discussion by Runaho** [dotnet/aspnetcore#49330] [Show and tell]
Hello,
After watching https://youtu.be/hVdwb41FPvU?t=3568 video and @captainsafia showcase.
I tried to run aspnetcore fork on my codespace but i got an error like the video 😄 .

Changes on made https://github.com/dotnet/aspnetcore/blob/main/.devcontainer/devcontainer.json file.

First i tried to run it root user ;
```json
  "remoteUser": "root"
```
Then i got another error and looked some issues i found this;
https://github.com/microsoft/vscode-remote-release/issues/8585#issuecomment-1585670432

After that i changed the onCreateCommand to this;
```json
  "onCreateCommand": "bash -i /workspaces/aspnetcore/.devcontainer/scripts/container-creation.sh",
```

I was able to run the container without any error.
Is this something that should happen, configured for local development or Is it just me?

**Comment by Runaho** [comment]
When i tried to reproducing this issue;
I only changed remoteUser to root & fixed the issue.

there is the steps for reproducing this issue;

- Create a fork & create codespace on main branch
<img width="437" alt="image" src="https://github.com/dotnet/aspnetcore/assets/16222645/71d82e4d-d625-44a9-bfca-723681474537">

First error log [creation-1.log](https://github.com/dotnet/aspnetcore/files/12026504/creation-1.log)

- After the i got this error i changed devcontainer.json file remoteUser to root.

I did this because I noticed errors such as downloading files, creating folders, insufficient disk.
<img width="1190" alt="image" src="https://github.com/dotnet/aspnetcore/assets/16222645/24801a9b-879c-4ddb-aa7d-e98484e9cf83">

**Comment by captainsafia** [comment]
Based on the error message, it appears the the issue might be related to the size of disk the machine is one. I'm surprised by this because the default size (32gb) seems like more than enough... It's not entirely clear to me why setting the `remoteUser` would resolve things either...

FWIW, I was able to repro this as well.

```
2023-07-19 01:07:16.242Z: /workspaces/aspnetcore/eng/targets/Npm.Common.targets(46,5): error : /workspaces/aspnetcore/src/SignalR/clients/ts/FunctionalTests/node_modules/puppeteer: Command failed. [/workspaces/aspnetcore/src/SignalR/clients/ts/FunctionalTests/SignalR.Npm.FunctionalTests.npmproj]
YARN : error : Failed to set up Chromium r1095492! Set "PUPPETEER_SKIP_DOWNLOAD" env variable to skip download. [/workspaces/aspnetcore/src/SignalR/clients/ts/FunctionalTests/SignalR.Npm.FunctionalTests.npmproj]
/workspaces/aspnetcore/eng/targets/Npm.Common.targets(46,5): error : Could not write file "/workspaces/aspnetcore/src/JSInterop/Microsoft.JSInterop.JS/src/yarn-error.log": "ENOSPC: no space left on device, write" [/workspaces/aspnetcore/src/JSInterop/Microsoft.JSInterop.JS/src/Microsoft.JSInterop.JS.npmproj]
/workspaces/aspnetcore/eng/targets/Npm.Common.targets(46,5): error : An unexpected error occurred: "ENOSPC: no space left on device, copyfile '/home/vscode/.cache/yarn/v6/npm-@nodelib-fs-stat-2.0.5-5bd262af94e9d25bd1e71b05deed44876a222e8b-integrity/node_modules/@nodelib/fs.stat/package.json' -> '/workspaces/aspnetcore/src/JSInterop/Microsoft.JSInterop.JS/src/node_modules/@nodelib/fs.stat/package.json'". [/workspaces/aspnetcore/src/JSInterop/Microsoft.JSInterop.JS/src/Microsoft.JSInterop.JS.npmproj]
/workspaces/aspnetcore/eng/targets/Npm.Common.targets(54,5): error : Could not write file "/workspaces/aspnetcore/src/JSInterop/Microsoft.JSInterop.JS/src/yarn-error.log": "ENOSPC: no space left on device, write" [/workspaces/aspnetcore/src/JSInterop/Microsoft.JSInterop.JS/src/Microsoft.JSInterop.JS.npmproj]
/workspaces/aspnetcore/eng/targets/Npm.Common.targets(54,5): error : An unexpected error occurred: "ENOSPC: no space left on device, copyfile '/home/vscode/.cache/yarn/v6/npm-@nodelib-fs-stat-2.0.5-5bd262af94e9d25bd1e71b05deed44876a222e8b-integrity/node_modules/@nodelib/fs.stat/package.json' -> '/workspaces/aspnetcore/src/JSInterop/Microsoft.JSInterop.JS/src/node_modules/@nodelib/fs.stat/package.json'". [/workspaces/aspnetcore/src/JSInterop/Microsoft.JSInterop.JS/src/Microsoft.JSInterop.JS.npmproj]
    0 Warning(s)
    6 Error(s)
```
